src/navigation/OccupancyGridClient.js

```diff
@@ -69,11 +69,13 @@ ROS3D.OccupancyGridClient.prototype.processMessage = function(message){
   // check for an old map
   if (this.currentGrid) {
     // check if it there is a tf client
-    if (this.currentGrid.tfClient) {
+    if (this.tfClient) {
       // grid is of type ROS3D.SceneNode
-      this.currentGrid.unsubscribeTf();
+      this.sceneNode.unsubscribeTf();
+      this.sceneNode.remove(this.currentGrid);
+    } else {
+      this.rootObject.remove(this.currentGrid);
     }
```

Comment:
Seems to me that, in order not to confuse people experience with Three.js coming into ros3d, we should just use the same object regardless if there is a TFClient or not, this way they don't have to check if OccupancyGrid has a client in order to further branch their own code so as to either access `sceneNode` or `rootObject`.

Or in other words, there should just be `sceneNode` and if there is no TFClient then it defaults to an empty one the emits nothing, or similar (this is something I do in downstrea code consuming ros3d at the moment, making a "MockTFClient" and passing that in instead, but something like this can be a default, or the code can be updated to accept null).

After such a change, the code would be this:

```js
    this.sceneNode.unsubscribeTf(); // no-op if no tfclient was provided.
    this.sceneNode.remove(this.currentGrid);
```

or (less preferably IMO):

```js
    if (this.sceneNode.tfClient) this.sceneNode.unsubscribeTf();
    this.sceneNode.remove(this.currentGrid);
```

Reply:
Interestingly, now that I realize it, this error would not exist if it was done that way to start with because one part of the code would not have made the assumption that the tree node was always the same.

Reply:
Umm. The question of review is does this patch fix the bug. I think it does and does so sufficiently. Your comment seems to trascend this patch and in and of itself may cause more regressions than the approach used here.